assert/assertions.go

```diff
@@ -529,14 +529,15 @@ func validateEqualArgs(expected, actual interface{}) error {
 //
 //	assert.Same(t, ptr1, ptr2)
 //
-// Both arguments must be pointer variables. Pointer variable sameness is
+// Both arguments must be pointer variables or something directly coercible
+// to a pointer such as a map or channel. Pointer variable sameness is
```

Comment:
What's coercible mean? Just say "Both arguments must be of kind pointer, map or channel".